src/main/java/com/amazon/ion/bytecode/bin11/bytearray/ShortTimestampDecoder.kt

```diff
@@ -0,0 +1,221 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.ion.bytecode.bin11.bytearray
+
+import com.amazon.ion.Timestamp
+import com.amazon.ion.bytecode.bin11.bytearray.PrimitiveDecoder.readFixedInt16
+import com.amazon.ion.bytecode.bin11.bytearray.PrimitiveDecoder.readFixedInt32
+import com.amazon.ion.bytecode.bin11.bytearray.PrimitiveDecoder.readFixedInt8AsShort
+import com.amazon.ion.bytecode.bin11.bytearray.PrimitiveDecoder.readFixedIntAsLong
+import java.math.BigDecimal
+
+/**
+ * Helper class for decoding the various short timestamp encoding variants from a [ByteArray].
```

Comment:
Small TODO note for later.

```suggestion
 * Helper class for decoding the various short timestamp encoding variants from a [ByteArray].
 *
 * TODO(perf): avoid auto-boxing the `0` integer for the offset when constructing the Timestamp instance.
```